src/Modal/Modal.tsx

```diff
@@ -41,20 +55,32 @@ const Modal: React.RefForwardingComponent<HTMLDivElement, ModalProps> = (
       const { top, left, width, height } = currentAnchor.getBoundingClientRect()
       setSize([top, left, width, height])
     }
-  }, [isOpen])
+    if (fullSize) {
+      setSize([0, 0, window.innerWidth - margin, window.innerHeight])
+    }
+  }, [isOpen, anchor, fullSize])
+
+  // Resize
+  React.useEffect(() => {
+    if (isOpen && fullSize) {
+      const handleResize = () => setSize([0, 0, window.innerWidth - margin, window.innerHeight])
+      window.addEventListener("resize", handleResize)
+      return () => window.removeEventListener("resize", handleResize)
+    }
+  }, [isOpen, fullSize])
```

Comment:
```suggestion
  }, [isOpen, fullSize, setSize])
```

(I know it's the same, but I would like to introduce the eslint hook rule later, so if we can avoid to much works over there ^^)